django/db/models/sql/query.py

```diff
@@ -2211,40 +2211,25 @@ def set_group_by(self, allow_aliases=True):
         primary key, and the query would be equivalent, the optimization
         will be made automatically.
         """
-        if allow_aliases:
-            # Column names from JOINs to check collisions with aliases.
-            column_names = set()
-            seen_models = set()
-            for join in list(self.alias_map.values())[1:]:  # Skip base table.
-                model = join.join_field.related_model
-                if model not in seen_models:
-                    column_names.update(
-                        {field.column for field in model._meta.local_concrete_fields}
-                    )
-                    seen_models.add(model)
-            if self.values_select:
-                # If grouping by aliases is allowed assign selected values
-                # aliases by moving them to annotations.
-                group_by_annotations = {}
-                values_select = {}
-                for alias, expr in zip(self.values_select, self.select):
-                    if isinstance(expr, Col):
-                        values_select[alias] = expr
-                    else:
-                        group_by_annotations[alias] = expr
-                self.annotations = {**group_by_annotations, **self.annotations}
-                self.append_annotation_mask(group_by_annotations)
-                self.select = tuple(values_select.values())
-                self.values_select = tuple(values_select)
+        if allow_aliases and self.values_select:
+            # If grouping by aliases is allowed assign selected value aliases
+            # by moving them to annotations.
+            group_by_annotations = {}
+            values_select = {}
+            for alias, expr in zip(self.values_select, self.select):
+                if isinstance(expr, Col):
+                    values_select[alias] = expr
+                else:
+                    group_by_annotations[alias] = expr
+            self.annotations = {**group_by_annotations, **self.annotations}
+            self.append_annotation_mask(group_by_annotations)
+            self.select = tuple(values_select.values())
+            self.values_select = tuple(values_select)
         group_by = list(self.select)
         for alias, annotation in self.annotation_select.items():
             if not (group_by_cols := annotation.get_group_by_cols()):
                 continue
-            if (
-                allow_aliases
-                and alias not in column_names
-                and not annotation.contains_aggregate
-            ):
```

Comment:
I think we need to revert changes in the `set_group_by()` to fix failures on Oracle.

Reply:
Latest commits should address that.